### PR TITLE
replaced some generic argument names with more descriptive names

### PR DESCRIPTION
**Pull request by turtleisaac**
As part of the commit contained in this pull request, I have replaced generic argument names (i.e. "arg0," "arg1,", etc...) for field scripting commands with more descriptive names that reflect what the value of the argument means.

These changes mostly relate to commands which involve overworld objects, but also include a trainer battle goto_if command. The changes made were according to the [Kingdom of DS Hacking Scrcmd Document](https://docs.google.com/spreadsheets/d/1rcrCz9Y5HXdD9vVSmkACX19I9Jid4dzFTD0PXaIDg2w/edit#gid=1350726349) and my own personal knowledge of what arguments these commands should take, along with context for the `goto_if_trainer_defeated` command since all of the other `goto_if` commands in Script.inc use `dest` as their argument name for the function to jump to.

For the `apply_movement` command, I used `sub` as the destination name, as it is more akin to one of the `call` or `call_if` commands, as execution returns to this point once the movement is started in parallel.

### CODE REVIEW

**Comment by red031000** [comment]
yeah we're not using `overworld` for events, it's very very poorly descriptive

**Comment by red031000** [comment]
diamond (and likely HG, not sure if I ported yet) uses LocalMapObject for events, much better imo than Overworld, cause that really does not mean anything, at least LocalMapObject is descriptive to what it is

**Comment by adrienntindall** [comment]
> diamond (and likely HG, not sure if I ported yet) uses LocalMapObject for events, much better imo than Overworld, cause that really does not mean anything, at least LocalMapObject is descriptive to what it is

Ditto on this, I understand where the terminology 'Overworld' comes from but Object is a much more common term outside of the rom hacking community.

**Comment by PikalaxALT** [comment]
Agree with the above, suggest `objectId`.

File: asm/macros/script.inc [hunk 1]
```diff
@@ -590,10 +590,10 @@
 	.short 93
 	.endm
 
-	.macro apply_movement arg0, arg1
+	.macro apply_movement overworld, sub
```
Comment (red031000) [inline asm/macros/script.inc:593]: delet overworld (see comments for suggestions)

**Review by red031000 — CHANGES_REQUESTED** [review]

**Comment by PikalaxALT** [comment]
Closing due to inactivity

**Pull request closed without merging**